internal/scalibrextract/language/java/pomxmlnet/extractor.go

```diff
@@ -69,7 +69,7 @@ func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]
 		}
 	}
 	// Merging parents data by parsing local parent pom.xml or fetching from upstream.
-	if err := mavenutil.MergeParents(ctx, e.MavenRegistryAPIClient, &project, project.Parent, 1, input.Path, true); err != nil {
+	if err := mavenutil.MergeParents(ctx, e.MavenRegistryAPIClient, &project, project.Parent, 1, filepath.Join(input.Root, input.Path), true); err != nil {
```

Comment:
shall we apply the change to other places calling `mavenutil.MergeParents`?

Reply:
I had a look at all the other instances of MergeParents, and none of them need this join, as they get the full path already (or are just empty).

It does look like we need to update the methods in osv-scanner fix to not depend on lockfile and move to the new interface.